pyproject.toml

```diff
@@ -1,3 +1,137 @@
+    "B024",  # abstract base class, but it has no abstract methods or properties
+    "B027",  # empty method in an abstract base class, but has no abstract decorator
+    "B028",  # No explicit `stacklevel` keyword argument found
+    "B904",  # Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
+    "D100",  # Missing docstring in public module  # FIXME: enable again
+    "D101",  # Missing docstring in public class  # FIXME: enable again
+    "D102",  # Missing docstring in public method  # FIXME: enable again
+    "D103",  # Missing docstring in public function  # FIXME: enable again
+    "D104",  # Missing docstring in public package  # FIXME: enable again
+    "D105",  # Missing docstring in magic method
+    "D106",  # Missing docstring in public nested class
+    "D107",  # Ignore D107 Missing docstring  # FIXME: enable again???
+    "D200",  # One-line docstring should fit on one line
+    "D202",  # No blank lines allowed after function docstring (found 1)
+    "D203",  # Ignore one-blank-line-before-class as Black handles this
+    "D204",  # 1 blank line required after class docstring
+    "D205",  # 1 blank line required between summary line and description
+    "D209",  # Multi-line docstring closing quotes should be on a separate line
+    "D210",  # No whitespaces allowed surrounding docstring text
+    "D211",  # No blank lines allowed before class docstring
+    "D212",  # Multi-line docstring summary should start at the first line
+    "D213",  # Multi-line docstring summary should start at the second line
+    "D301",  # Use `r"""` if any backslashes in a docstring
+    "D400",  # First line should end with a period
+    "D401",  # First line of docstring should be in imperative mood
+    "D402",  # First line should not be the function's signature
+    "D404",  # First word of the docstring should not be "This"
+    "D415",  # First line should end with a period, question mark, or exclamation point
+    "E226",  # Missing whitespace around arithmetic operator  # FIXME: enable again
+    "E241",  # Multiple spaces after comma  # FIXME: enable again
+    "I001",  # Import block is un-sorted or un-formatted  # FIXME: enable again
+]
+preview = true
+
+[tool.ruff.lint.per-file-ignores]
+"*/__init__.py" = ["F401"]
+"example/*" = ["F403", "F405"]
+
+[tool.mypy]
+files = ["shc/", "test/", "example/"]
+disallow_untyped_decorators = true
+warn_unreachable = true
+check_untyped_defs = true
+namespace_packages = false
+plugins = ["shc/util/mypy_variable_plugin.py"]
+exclude = "^test/assets/mypy_plugin_test"
+
+[[tool.mypy.overrides]]
+module = "mido,astropy,astropy.*,serial,serial_asyncio,paho.mqtt.*,selenium.*,pulsectl.*,pulsectl_asyncio,aiomysql,aiogram.*"
+ignore_missing_imports = true
+
+[tool.ruff.format]
+# Like Black, use double quotes for strings.
+# quote-style = "double"  # FIXME: do we want this now?
```

Comment:
For me it's fine to always have quotes.  But we can discuss it once we address formatting in the 2nd stage.